provider/pkg/provider/provider.go

```diff
@@ -419,10 +421,17 @@ func (k *kubeProvider) DiffConfig(ctx context.Context, req *pulumirpc.DiffReques
 	// The alternative of ignoring changes to the kubeconfig is untenable; if the k8s cluster has
 	// changed, any dependent resources must be recreated, and ignoring changes prevents that from
 	// happening.
-	oldActiveCluster := getActiveClusterFromConfig(oldConfig, olds)
-	activeCluster := getActiveClusterFromConfig(newConfig, news)
-	if !reflect.DeepEqual(oldActiveCluster, activeCluster) {
-		replaces = diffs
+	oldActiveCluster, oldFound := getActiveClusterFromConfig(oldConfig, olds)
+	activeCluster, found := getActiveClusterFromConfig(newConfig, news)
+	if !oldFound || !found {
+		// The config is either ambient or invalid, so we can't draw any conclusions.
+	} else if !reflect.DeepEqual(oldActiveCluster, activeCluster) {
+		// one of these properties must have changed for the active cluster to change.
+		for _, key := range []string{"kubeconfig", "context", "cluster"} {
+			if slices.Contains(diffs, key) {
+				replaces = append(replaces, key)
+			}
+		}
```

Comment:
This change makes the replacement logic be slightly less conservative, because it now won't suggest a replacement when the old or new cluster info couldn't be resolved, e.g. when the config is ambient or the `context` doesn't match anything. Given that this PR fixes a problem in `parseKubeconfigPropertyValue`, we're more likely to reach this logic than in the past.